### ISSUE TITLE
pytz.exceptions.AmbiguousTimeError during DST changes if created with choices_display='WITH_GMT_OFFSET'

### ISSUE THREAD

**Issue by atleta** [mfogel/django-timezone-field#72]
Whenever you try to initialize a model with a TimeZone field during an a DST change in *any* of the timezones supported by pytz you'll get the following exception:

```

 File ".../devel/backend/env/lib/python3.7/site-packages/timezone_field/fields.py", line 72, in __init__                                                          
    choices = with_gmt_offset(values)                                                                                                                                                                      
  File ".../devel/backend/env/lib/python3.7/site-packages/timezone_field/choices.py", line 34, in with_gmt_offset                                                  
    delta = pytz.timezone(tz_str).utcoffset(now)                                                                                                                                                           
  File ".../devel/backend/env/lib/python3.7/site-packages/pytz/tzinfo.py", line 422, in utcoffset                                                                  
    dt = self.localize(dt, is_dst)                                                                                                                                                                         
  File ".../devel/backend/env/lib/python3.7/site-packages/pytz/tzinfo.py", line 363, in localize                                                                   
    raise AmbiguousTimeError(dt)                                                                                                                                                                           
pytz.exceptions.AmbiguousTimeError: 2021-04-03 23:44:14.162132
```

To test this you'd probably want to use freezegun (or something similar) with a unit test. You can use the above timestamp as an example. (It should produce an error when evaluating the `America/Santiago` timezone.)

Now looking into your code (`with_gmt_offset` in `choices.py`) the logic doesn't even seem valid. `pytz.timezone(...).utcoffset` will tell you the UTC offset of the provided timestamp. I.e. the utc offset of the specified timezone *at provided timestamp*. But you are passing in the utctimestamp and this will provide a bogus result around the DST (even if you do fix the above error). Because it will tell you the utc offset not for the current time but for the time returned by utcnow. Which will be offset by the utc offset itself... So e.g. I'm in Europe, which means I'm UTC+1 (without DST), so when I get to the DST change (transitioning to UTC+2) your algorithm will show me I'm in UTC+1 for until an hour after the actual DST change.

Now I think the whole algoritm could be a lot simpler. You just create the timezone instances, then you create localized datetimes and get the utcoffset for those. This removes the whole issue (the guesswork that pytz has to do).

Because then you go UTC -> localized timestamp, which is unambiguous. The localized timestamp (based on the UTC) will contain the correct timezone (either DST or non-DST) and you can get the offset from that without any guesswork. (Again, the guesswork is hidden in pytz.)

**Comment by mfogel** [comment]
thanks for the bug report @atleta 

What would be super helpful if you have the time is to write up a failing test case and add it to https://github.com/mfogel/django-timezone-field/blob/master/tests/tests.py#L485-L575. Either as a PR or just a block of code on this ticket. If you don't have time I will get to this as soon as I can, but it'll be a few days at best

#71 is probably the same issue

**Comment by atleta** [comment]
@mfogel I've looked into the code on github and it turns out that you have already fixed it here: https://github.com/mfogel/django-timezone-field/blob/72bb56fbde1afdac507a77f4e65316dc7c18a595/timezone_field/choices.py#L33

And this seems to fix the issue (I haven't tested it)

Interesting enough, when testing the bogus (4.1.1) version in the console with a trivial test case (see below) it gives the NonExistentTimeError. I.e. a different one than what I have seen when trying to run e.g. `manage shell` (or `runserver`). But the issue was the same.

```
with freeze_time('2021-03-28 01:44:14.162132'):
    TimeZoneField(default='UTC', choices_display='WITH_GMT_OFFSET')
```

or even 

```
with freeze_time('2021-03-28 01:44:14.162132'):
     class M(models.Model):
        TimeZoneField(default='UTC', choices_display='WITH_GMT_OFFSET')
```

Anway, regarding your fix (see above): the replace and the substraction is unnecessary. You could simply use: `delta = now_tz.utcoffset()`. It won't cause a problem due to what I have said above.

**Comment by mfogel** [comment]
> Anway, regarding your fix (see above): the replace and the subtraction is unnecessary. You could simply use: delta = now_tz.utcoffset(). 

Please feel free to give it a go! Make sure the tests still pass, and if they do, then please file a pull request.

Sounds like this bug is a duplicate of #70 which affected `v4.1.1` and was fixed in 37249c6 for `v4.1.2`. If that's not right and you are seeing this bug with version `v4.1.2` then please reopen.
